cargo-dist/src/init/v0.rs

```diff
@@ -148,38 +130,26 @@ pub fn do_init(cfg: &Config, args: &InitArgs) -> DistResult<()> {
     // ctrl-c handler.
     ctrlc_handler.abort();
 
-    // If we're migrating, the configuration will be missing the
-    // generic workspace specification, and will have some
-    // extraneous cargo-specific stuff that we don't want.
-    let mut workspace_toml = if newly_initted_generic {
-        migrate::new_cargo_workspace()
-    } else {
-        workspace_toml
-    };
+    let root_workspace = workspaces.root_workspace();
+
+    // Load in the root workspace toml to edit and write back
+    let mut workspace_toml = config::load_toml(&root_workspace.manifest_path)?;
 
     if let Some(meta) = &multi_meta.workspace {
-        apply_dist_to_workspace_toml(&mut workspace_toml, desired_workspace_kind, meta);
+        apply_dist_to_workspace_toml(&mut workspace_toml, root_workspace.kind, meta);
     }
 
     eprintln!();
 
-    let filename;
-    let destination;
-    if newly_initted_generic {
-        // Migrations and newly-initted setups always use dist-workspace.toml.
-        filename = "dist-workspace.toml";
-        destination = root_workspace.workspace_dir.join(filename);
-    } else {
-        filename = root_workspace
-            .manifest_path
-            .file_name()
-            .expect("no filename!?");
-        destination = root_workspace.manifest_path.to_owned();
-    };
+    let filename = root_workspace
```

Comment:
ahhh so much nicer

Reply:
as a fun aside, this has actually been unneeded since we started having `dist init` call `dist migrate`. i just wasn't as confident of it until i was more familiar with this part of the codebase.